lib/generators/blueprinter/initializer_generator.rb

```diff
@@ -0,0 +1,56 @@
+module Blueprinter
+  module Generators
+    class InitializerGenerator < Rails::Generators::Base
+      desc "Generates an initializer for Blueprinter"
+
+      attr_accessor :options
+
+      source_root File.expand_path("../templates", __FILE__)
+
+
+
+      class_option :initializer_dir, default: "config/initializers", desc: "path to initializer file"
+
+      class_option :name, default: "blueprinter.rb", desc: "name of initializer file"
+
+
+
+      class_option :generator, default: nil, desc: "What gem to use for JSON", banner: "oj|yajl"
+
+      class_option :method, default: nil, desc: "What method to call on the generator", banner: "encode"
+
+
+
+      class_option :sort_fields_by, default: nil, desc: "How to sort JSON fields", banner: "definition"
+
+
+
+      class_option :field_default, type: :string, default: nil, desc: "field_default config option", banner: "\"N/A\""
+
+      class_option :association_default, type: :string, default: nil, desc: "association_default config option", banner: "{}", aliases: "-a"
+
+
+
+      remove_class_option :skip_namespace
+
+      def ensure_initializer_dir
+        FileUtils.mkdir_p(options["initializer_dir"]) unless File.directory?(options["initializer_dir"])
+      end
+
+      def create_initializer
+        template "initializer.rb", File.join(options["initializer_dir"], options["name"] )
+      end
+
+      private
+
+      def generator_gem
+        {oj: "Oj", yajl: "Yajl::Encoder"}[options["generator"].intern] if options["generator"]
+      end
+
+      def empty_string(default)
+        default.to_s.gsub(/"'/,'').blank?
+      end
+
+    end
+  end
+end
```

Comment:
Would you make a separate PR for the initializers generator? Reasoning is that I would like to have a more thorough discussion regarding generating initializers and the default configurations, and I don't want to have that block the rest of this PR. Currently, I think it's not very useful to generate an empty initializer by default.
```ruby
Blueprinter.configure do |config|
end
```

Rather I'd like to have the initializer generator to generate all the defaults configs of blueprinter, with code comments showing potential options.
Something like
```ruby
Blueprinter.configure do |config|
  # The default generator is JSON. You may replace it with OJ or Yajil as well.
  config.generator = JSON
end
```

Reply:
sounds good. I took that part out. I'll wait for the next pr until this is ready to merge in case I drop off the internet again ;)